include/rmm/mr/device/detail/arena.hpp

```diff
@@ -647,29 +647,33 @@ class global_arena final {
    *
    * @param logger the spdlog logger to use
    */
+#ifdef RMM_BACKWARDS_COMPATIBILITY
   RMM_HIDDEN void dump_memory_log(std::shared_ptr<spdlog::logger> const& logger) const
+#else
+  void dump_memory_log(std::shared_ptr<logger> const& logger) const
+#endif
   {
     std::lock_guard lock(mtx_);
 
-    logger->info("  Arena size: {}", rmm::detail::bytes{upstream_block_.size()});
```

Comment:
I didn't realize that we have to get rid of all uses of `format` placeholders. I would have thought that since these are inside of string literals they wouldn't expose anything from `fmt` publicly.

Reply:
Paraphrasing something I wrote offline: in principle I think it would be possible for us to keep this functionality and find a way to hide enough details of the fmt function calls to not leak anything, but it is tricky, especially given that rmm is header-only so its includes propagate to consumers, who then have similar responsibilities. The problem isn't these strings themselves, but the functions that ultimately process these strings (which must be variadic templates and therefore must be implemented, not just declared, in header files that expose the internals). I think we're better off losing some quality of debug print until we upgrade to C++20 to simplify package maintenance in the long run.

Reply:
Looks like spdlog / fmt require `{}` placeholders. I'd like to see an example of where a format string containing `{}` placeholders passed to spdlog exposes `fmt` symbols but a format string containing `%p`/`%s`/etc placeholders does not.